### DISCUSSION TITLE
xtts api server help

### DISCUSSION THREAD

**Discussion by texMachina33** [lukaszliniewicz/Pandrator#58] [Q&A] [answered]
I don't know how to start the XTTS API Server after I extract from Pandrator_XTTS_RVC_Whisperx_Training_v026 and run the installer.
if someone could give me a resource to understand where I need to run _conda activate xtts_api_server_installer
python -m xtts_api_server_ I would be incredibly grateful.

**Comment by lukaszliniewicz** [comment]
When you unpack the archive, there will be an exe file - pandrator_installer_launcher. You may use it to start the XTTS server and Pandrator. 

![image](https://github.com/user-attachments/assets/bccc8eda-79a4-48d5-b6a3-2f9fd3a20439)

Version 0.26 may contain a bug that prevents Pandrator from launching. Please make sure that you have the **[latest version](https://1drv.ms/f/s!AgSiDu9lV3iMnPFKPO5BB_c72OLjtQ?e=3fRZMG)**. 

The commands with `conda activate` were for a manual installation. 

If you'd rather run XTTS and Pandrator manually after unpacking the archive, execute this command from inside the Pandrator\xtts_api_server folder:

`..\conda\Scripts\conda.exe run -p ..\conda\envs\xtts_api_server_installer\ --no-capture-output python -m xtts_api_server --deepspeed
`

For Pandrator, from the Pandrator folder, in a different powershell session, run

`.\conda\Scripts\conda.exe run -p .\conda\envs\pandrator_installer\ --no-capture-output .\Pandrator\pandrator.py
`

**Reply by texMachina33** [reply]
ill try a fresh install. i had (v26.7) but will it work with my amd gpu.

**Comment by texMachina33** [comment]
i also unpacked the archive and ran from the installer.exe in the folder. this is the console when i run it. a copy of the log below.
Waiting for xtts server to come online...
Attempt 1/200: xtts server is not online yet. Waiting...
XTTS stderr: Traceback (most recent call last):
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\conda\envs\xtts_api_server_installer\lib\site-packages\transformers\__init__.py", line 4370, in <module>
XTTS stderr: raise OptionalDependencyNotAvailable()
XTTS stderr: transformers.utils.import_utils.OptionalDependencyNotAvailable
XTTS stderr:
XTTS stderr: During handling of the above exception, another exception occurred:
XTTS stderr:
XTTS stderr: Traceback (most recent call last):
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\conda\envs\xtts_api_server_installer\lib\runpy.py", line 196, in _run_module_as_main
XTTS stderr: return _run_code(code, main_globals, None,
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\conda\envs\xtts_api_server_installer\lib\runpy.py", line 86, in _run_code
XTTS stderr: exec(code, run_globals)
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\xtts-api-server\xtts_api_server\__main__.py", line 46, in <module>
XTTS stderr: from xtts_api_server.server import app
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\xtts-api-server\xtts_api_server\server.py", line 18, in <module>
XTTS stderr: from xtts_api_server.tts_funcs import TTSWrapper,supported_languages,InvalidSettingsError
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\xtts-api-server\xtts_api_server\tts_funcs.py", line 8, in <module>
XTTS stderr: from TTS.tts.configs.xtts_config import XttsConfig
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\conda\envs\xtts_api_server_installer\lib\site-packages\TTS\tts\configs\xtts_config.py", line 5, in <module>
XTTS stderr: from TTS.tts.models.xtts import XttsArgs, XttsAudioConfig
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\conda\envs\xtts_api_server_installer\lib\site-packages\TTS\tts\models\xtts.py", line 11, in <module>
XTTS stderr: from TTS.tts.layers.xtts.gpt import GPT
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\conda\envs\xtts_api_server_installer\lib\site-packages\TTS\tts\layers\xtts\gpt.py", line 9, in <module>
XTTS stderr: from transformers import GPT2Config
XTTS stderr: File "C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\conda\envs\xtts_api_server_installer\lib\site-packages\transformers\__init__.py", line 4372, in <module>
XTTS stderr: from .utils import (
XTTS stderr: ImportError: cannot import name 'dummy_essentia_and_librosa_and_pretty_midi_and_scipy_and_torch_objects' from 'transformers.utils' (C:\Users\devin\OneDrive\Desktop\pandarator startup base files\Pandrator_XTTS_RVC_Whisperx_Training_v026\Pandrator\conda\envs\xtts_api_server_installer\lib\site-packages\transformers\utils\__init__.py)
XTTS stderr: ERROR conda.cli.main_run:execute(125): `conda run python -m xtts_api_server --device cpu` failed. (See above for error)
Attempt 2/200: xtts server is not online yet. Waiting...
Attempt 3/200: xtts server is not online yet. Waiting...
Attempt 4/200: xtts server is not online yet. Waiting...
Attempt 5/200: xtts server is not online yet. Waiting...
Attempt 6/200: xtts server is not online yet. Waiting...
Attempt 7/200: xtts server is not online yet. Waiting...
Attempt 8/200: xtts server is not online yet. Waiting...
Attempt 9/200: xtts server is not online yet. Waiting...
Attempt 10/200: xtts server is not online yet. Waiting...
it continued like this till 40 attempts when i closed it. I'm thinking its caused by my amd gpu/cpu even though i used xtts use cpu. granted I'm not an expert.
if it helps i got a sapphire nitro +7900xtx gpu.

**Comment by texMachina33** [comment, marked as answer]
the easiest fix I've found is to just use silero. thank you for the help lukaszliniewicz. I truly do appreciate it.

**Comment by lukaszliniewicz** [comment]
I will upload a package with just CPU support for XTTS. You have an AMD GPU, which isn't supported by default, and I suspect the problem is that it's trying to initialise RVC on launch (it comes installed with the largest package). The Pandrator_XTTS package should work, but to be sure I will make one without CUDA support, it will be smaller as well. Sorry for the problems.

**Comment by lukaszliniewicz** [comment]
I've uploaded a CPU-only version of the XTTS package: https://1drv.ms/f/s!AgSiDu9lV3iMnPFKPO5BB_c72OLjtQ?e=3fRZMG

**Reply by texMachina33** [reply]
thank you
